### PR TITLE
Migrate dashboards to by_value

### LINKED ISSUE #3463: [security-external-integrations packages] Migration dashboards to by_value
Many of the dashboards include references to visualizations. We want to include the visualizations by value in dashboards such that it minimizes the saved object clutter and reduces time to load. These are the integrations that require conversions.

A migration script was created to help with this effort: https://github.com/flash1293/legacy_vis_analyzer

Relates: #3448

|   Integration    	|   By Reference Count 	|
|--------------------------------	|-------------------------	|
|   1password                    	|   13                    	|
|   auditd                       	|   6                     	|
|   auditd_manager               	|   5                     	|
|   auth0                        	|   2                     	|
|   carbon_black_cloud           	|   69                    	|
|   cef                          	|   71                    	|
|   cisco                        	|   10                    	|
|   cisco_asa                    	|   10                    	|
|   cisco_duo                    	|   25                    	|
|   cisco_ise                    	|   61                    	|
|   cisco_secure_email_gateway   	|   54                    	|
|   cloudflare                   	|   78                    	|
|   cyberarkpas                  	|   2                     	|
|   gcp                          	|   1                     	|
|   github                       	|   12                    	|
|   hashicorp_vault              	|   1                     	|
|   hid_bravura_monitor          	|   123                   	|
|   infoblox_nios                	|   24                    	|
|   iptables                     	|   17                    	|
|   microsoft                    	|   7                     	|
|   microsoft_defender_endpoint  	|   7                     	|
|   mimecast                     	|   12                    	|
|   netflow                      	|   79                    	|
|   netskope                     	|   88                    	|
|   network_traffic              	|   116                   	|
|   o365                         	|   7                     	|
|   okta                         	|   6                     	|
|   osquery                      	|   12                    	|
|   panw                         	|   15                    	|
|   pfsense                      	|   26                    	|
|   proofpoint_tap               	|   29                    	|
|   qnap_nas                     	|   6                     	|
|   santa                        	|   7                     	|
|   sentinel_one                 	|   1                     	|
|   sonicwall_firewall           	|   1                     	|
|   suricata                     	|   18                    	|
|   tenable_sc                   	|   20                    	|
|   zeek                         	|   8                     	|
|   zscaler_zia                  	|   31                    	|
|   zscaler_zpa                  	|   30                    	|

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?
As a best practice, this would move visualisations to by_value instead of by_reference

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues
- Closes #3463

### Commands used for an integration:
- `elastic-package stack down && elastic-package stack up --version=x.y.z -d -v` (x.y.z values taken from `kibana.version` in the package's `manifest.yml`)
- `cd legacy_vis_analyzer` (cloned from https://github.com/flash1293/legacy_vis_analyzer)
- `KIBANA="https://elastic:changeme@localhost:5601" NODE_TLS_REJECT_UNAUTHORIZED='0' node inline.js ./elastic/integrations/packages/1password/kibana/`

**Sample output:**
```
Prepared 8 visualizations to be inlined
Prepared 2 maps to be inlined
Processing dashboard Item Usages [1Password]
Leaving panel of type search
Inlined a map, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Processing dashboard Sign-in Attempts [1Password]
Leaving panel of type search
Inlined a map, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined a vis, pushed 1 inner references
Inlined 8 visualizations
Inlined 2 maps
Inlined 0 lenses
Removing visualization folder
Removing maps folder
Writing back dashboards

### CODE REVIEW

File: packages/auditd/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "3.3.5"
+  changes:
+    - description: Migrate dashboards to by_value
```
Comment (andrewkroh) [inline packages/auditd/changelog.yml:4]: Most end users will not understand what this means or how it impacts them. Please elaborate.

**Review by andrewkroh — COMMENTED** [review]
Assuming this was automated, please include the commands used to reproduce this in the commit messages.

I will often use https://pkg.go.dev/rsc.io/rf/git-generate to replay the automatic steps if I need to fix merge conflicts. So if you store the commands in the commit message then you can do this with ease.

File: packages/auditd/changelog.yml (hunk 1 above)
Comment (kcreddy) [reply packages/auditd/changelog.yml]: Elaborated the description to help users understand the change being made.

**Comment by kcreddy** [comment]
@andrewkroh I started working on script, but then noticed that the [conversion tool](https://github.com/flash1293/legacy_vis_analyzer) is changing the `migrationVersion` and `coreMigrationVersion` values inside the dashboard files. That would probably require additional changes to `kibana.version` property inside `manifest.yml`. But there may be customers who have been actively using minimum stack version, which seem to be different for different packages. So, I ran the [conversion tool ](https://github.com/flash1293/legacy_vis_analyzer) for each package manually using the minimum stack version as per respective `manifest.yml`

**Comment by andrewkroh** [comment]
> So, I ran the [conversion tool ](https://github.com/flash1293/legacy_vis_analyzer)for each package manually using the minimum stack version as per respective manifest.yml

Can you provide an example of the specific commands used for one integration migration. And put that into the PR description.

**Comment by kcreddy** [comment]
> > So, I ran the [conversion tool ](https://github.com/flash1293/legacy_vis_analyzer)for each package manually using the minimum stack version as per respective manifest.yml
> 
> Can you provide an example of the specific commands used for one integration migration. And put that into the PR description.

Added the commands and also sample output. One thing to note is that the tool didn't convert the `searches` into by value. They are still present as by reference.

**Comment by flash1293** [comment]
It looks like a bunch of these are fully inlined already (gcp, github), so inlining is not doing anything except bumping the version (basically in all the cases no visualizations/lens/maps saved objects got deleted in a package). Should we revert these? Otherwise the user might be confused that nothing at all is happening in the upgrade.

**Comment by kcreddy** [comment]
@flash1293 I also initially thought of not making any changes to those packages, but I noticed few parameters like `namespaces`, `updated_at`, `version` are getting added to the dashboard files by the conversion tool. So, maybe I thought keeping the change could help bringing the dashboard files to a common format across all packages.

But from the user perspective, I do believe that it could cause confusion. I reverted the changes for all such packages. Let me know if it looks good. Thanks!

**Comment by flash1293** [comment]
@kcreddy that's OK, it's just an artifact of the saved object migration and shouldn't matter.


There's this limitation of the inliner script at the moment:

> For old dashboards (prior to 7.10), some "agg based" visualizations might break if a target version of 7.17 or 8.0 is used. In these cases, please use at least a stack version of 8.1

It looks like you used it for some older Kibana versions (e.g. `1password`) - this isn't neccessarily a problem, but someone who knows the integration should actually run them and make sure the visualizations are still functioning as expected.

**Comment by kcreddy** [comment]
@flash1293 I upgraded all those packages to 8.1.0 as per the caveat mentioned in the tool. Dashboards were indeed not loading with errors

**Comment by kcreddy** [comment]
> Assuming you've done the manual testing the was mentioned above, LGTM.

@andrewkroh I had to upgrade the `kibana.version` to `^8.1.0` for these 4 packages so that the dashboards are not broken with the error Joe mentioned above. Let me know if thats okay.
`o365, okta, osquery, microsoft_defender_endpoint`

**Comment by kcreddy** [comment]
Taking a closer look at @flash1293 comment, it seems like I need to test loading visualisations for packages with older stack versions and not just on packages where I had to change the version due to errors from tool. Let me load the dashboards for all these older packages, could take sometime but worth doing to avoid any inconveniences for the customers.

**Comment by kcreddy** [comment]
Hi @andrewkroh, 
I had manually checked the dashboards for all these packages. As indicated by @flash1293, most of the older packages have the issue where the dashboards loads with an error `Trying to initialize aggs without index pattern`. So, I upgraded such packages to a minimum version of `8.1.0` and again manually tested them which are loading fine now.

Some of the packages' dashboards are converted into by_value by crest team already.

Here is the summary:

* `1password` **upgraded** to 8.1.0 as some agg vis failed to load
* `auditd` **upgraded** to 8.1.0 as some agg vis failed to load
* `Auth0` **upgraded** to 8.1.0 as some agg vis failed to load even in 7.17.7
* `carbon_black_cloud` reverted current change as done by crest
* `cef` **upgraded** to 8.1.0 as some agg vis failed to load
* `cisco` reverted current change as deprecated
* `cisco_asa` **upgraded** to 8.1.0 as some agg vis failed to load
* `cisco_duo` reverted as done by crest
* `cisco_ise` **upgraded** to 8.1.0 as some agg vis failed to load
* `cisco_secure_email_gateway` reverted current change as done by crest
* `Cloudflare` loads fine with current stack version
* `hid_bravura` **upgraded** to 8.1.0 as some agg vis failed to load
* `Iptables` **upgraded** to 8.1.0 as some agg vis failed to load
* `microsoft_defender_endpoint` **upgraded** to 8.1.0 as some agg vis failed to load
* `Mimecast` loads fine with current stack version
* `netflow` **upgraded** to 8.1.0 as some agg vis failed to load
* `netskope` **upgraded** to 8.1.0 as some agg vis failed to load
* `network_traffic` loads fine with current stack version
* `o365` **upgraded** to 8.1.0 as some agg vis failed to load
* `okta` **upgraded** to 8.1.0 as some agg vis failed to load
* `osquery` **upgraded** to 8.1.0 as some agg vis failed to load
* `pfsense` **upgraded** to 8.1.0 as some agg vis failed to load
* `qnap_nas` **upgraded** to 8.1.0 as some agg vis failed to load
* `santa` **upgraded** to 8.1.0 as some agg vis failed to load
* `suricata` **upgraded** to 8.1.0 as some agg vis failed to load
* `tenable_sc` reverted current change as done by crest
* `zeek` **upgraded** to 8.1.0 as some agg vis failed to load
* `zscalar_zia` reverted current change as done by crest
* `zscalar_zpa` reverted current change as done by crest

Considering these additional changes (`conditions.kibana.version` in manifest.yml), can I go ahead with merging the PR?

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**